test/csv/parse/test_row_separator.rb

```diff
@@ -13,4 +13,17 @@ def test_multiple_characters
                    CSV.parse("a\r\nb\r\n", row_sep: "\r\n"))
     end
   end
+
+  def test_skip_row_recovery_after_faulty_quoting_with_row_ending_with_cr_lf
+    data = <<-CSV
+"1","OK"\r
+"2",""NOT" OK"\r
+"3","OK"\r
+CSV
+    csv = CSV.new(data)
+
+    assert_equal('1',  csv.shift[0])
+    assert_raise(CSV::MalformedCSVError) { csv.shift }
+    assert_equal('3',  csv.shift[0])
```

Comment:
It's as if there's an additional space after the first string in the '1' and '3' assertions.

Reply:
Good catch. Should be fixed now.